aspnetcore/host-and-deploy/proxy-load-balancer/6.1samples/WebPS/Program.cs

```diff
@@ -0,0 +1,468 @@
+app.MapRazorPages();
+
+app.Run();
+#endregion
+
+public static class ConversionLogic
+{
+    public static System.Security.Cryptography.X509Certificates.X509Certificate2
+        CreateAnX509Certificate2() => new System.Security.Cryptography
+        .X509Certificates.X509Certificate2("abc");
+}
+#elif TRB
+#region snippet_trb
+using Microsoft.AspNetCore.HttpOverrides;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+builder.Services.Configure<ForwardedHeadersOptions>(options =>
+{
+    options.ForwardedHeaders =
+        ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
+});
+
+var app = builder.Build();
+
+app.UseForwardedHeaders();
+
+app.Run(async (context) =>
+{
+    context.Response.ContentType = "text/plain";
+
+    // Request method, scheme, and path
+    await context.Response.WriteAsync(
+        $"Request Method: {context.Request.Method}{Environment.NewLine}");
+    await context.Response.WriteAsync(
+        $"Request Scheme: {context.Request.Scheme}{Environment.NewLine}");
+    await context.Response.WriteAsync(
+        $"Request Path: {context.Request.Path}{Environment.NewLine}");
+
+    // Headers
+    await context.Response.WriteAsync($"Request Headers:{Environment.NewLine}");
+
+    foreach (var header in context.Request.Headers)
+    {
+        await context.Response.WriteAsync($"{header.Key}: " +
+            $"{header.Value}{Environment.NewLine}");
+    }
+
+    await context.Response.WriteAsync(Environment.NewLine);
+
+    // Connection: RemoteIp
+    await context.Response.WriteAsync(
+        $"Request RemoteIp: {context.Connection.RemoteIpAddress}");
+});
+
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseAuthorization();
+
+app.MapRazorPages();
+
+app.Run();
+#endregion
+#elif TRB2
+#region snippet_trb2
+using Microsoft.AspNetCore.HttpOverrides;
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+builder.Services.Configure<ForwardedHeadersOptions>(options =>
+{
+    options.ForwardedHeaders =
+        ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
+});
+
+var app = builder.Build();
+
+app.UseForwardedHeaders();
+
+ILogger _logger = app.Logger;
+
+app.Use(async (context, next) =>
+{
+    // Request method, scheme, and path.
+    _logger.LogInformation("Request Method: {Method}", context.Request.Method);
+    _logger.LogInformation("Request Scheme: {Scheme}", context.Request.Scheme);
+    _logger.LogInformation("Request Path: {Path}", context.Request.Path);
+
+    // Headers
+    foreach (var header in context.Request.Headers)
+    {
+        _logger.LogInformation("Header: {Key}: {Value}", header.Key, header.Value);
+    }
+
+    // Connection: RemoteIp
+    _logger.LogInformation("Request RemoteIp: {RemoteIpAddress}",
+        context.Connection.RemoteIpAddress);
+
+    await next();
+});
```

Comment:
This suggestion removes the `_` from `_logger` as it's a local variable. It also passes `context` into the call to `Use`:

```suggestion
var logger = app.Logger;

app.Use(async (context, next) =>
{
    // Request method, scheme, and path.
    logger.LogInformation("Request Method: {Method}", context.Request.Method);
    logger.LogInformation("Request Scheme: {Scheme}", context.Request.Scheme);
    logger.LogInformation("Request Path: {Path}", context.Request.Path);

    // Headers
    foreach (var header in context.Request.Headers)
    {
        logger.LogInformation("Header: {Key}: {Value}", header.Key, header.Value);
    }

    // Connection: RemoteIp
    logger.LogInformation("Request RemoteIp: {RemoteIpAddress}",
        context.Connection.RemoteIpAddress);

    await next(context);
});
```